design/ShaderLanguageFAQ.md

```diff
@@ -0,0 +1,157 @@
+
+
+## What does a WGSL shader look like?
+
+WGSL has not yet been specified.
+However, early prototyping work indicates the language can have a look
+and feel as represented in the following examples:
+
+A simple fragment shader:
+
+
+```
+[[location 0]] var<out> gl_FragColor : vec4<f32>;
+
+fn main() -> void {
+    gl_FragColor = vec4<f32>(0.4, 0.4, 0.8, 1.0);
+    return;
+}
+entry_point fragment = main;
+```
+
+
+A code fragment showing if/else:
+
+
+```
+  if (value > 5) {
+    wvar = 14;
+  } elseif (value > 2) {
+    wvar = 12;
+  } else {
+    wvar = 0;
+  }
+```
+
+
+A code fragment showing a switch:
+
+
+```
+switch (a) {
+  case 1: {
+    # The break is implicit
+  }
+  case 2: {
+     fallthrough;
+  }
+  default: {
+  }
+}
+```
+
+
+A code fragment showing a loop:
+
+
+```
+var i : i32 = 0;
+var b : f32 = 2.4;
+loop {
+  break if (i == 4);
+
+  b = b * 3.2;
+  continuing {
+     i = i + 1;
+  }
+}
+```
+
+
+
+## Where can I find the spec?
+
+The WGSL spec will be developed on the WebGPU Community Group area on GitHub.
+See https://github.com/gpuweb/gpuweb/tree/master/wgsl
+
+
+## What browsers support WGSL?
+
+WGSL is the shader language for WebGPU.
+Any browser implementing WebGPU will be required to accept shaders in WGSL.
+Specifically, the WebGPU conformance test suite will use shaders written
+in WGSL.
+
+Browser support for WGSL is expected to be developed over time, along
+with the development of WGSL itself.
+
+
+## Will browsers accept other shader languages for WebGPU?
+
+Accepting a shader language other than WGSL is not part of WebGPU.
+Doing so harms interoperability, as a shader in that other language is
+not portable across browsers.
+
+
+## Where can I find WGSL tooling?
+
+The WebGPU Community Group does not deliver tooling for WGSL.
+However, WebGPU implementers are expected to develop tooling
+for handling WGSL, including parsing, validation, and compilation.
+
+At this time, we expect at least three essentially independent
+implementations of WGSL ingestion as part of browser support for WebGPU.
+Independent implementation serves as an important check on the quality
+of the WGSL and WebGPU specifications.
+
+
+## What about my existing shaders, compilers, and other tooling?
+
+This is also expressed as “the last thing I need is another shader language”.
+(See https://xkcd.com/927/)
+
+For developers invested in the SPIR-V ecosystem for Vulkan and OpenGL,
+the key will be convertibility between WGLSL and SPIR-V.
+For shaders that only use WebGPU features, WebGPU Community Group
+members expect to develop tooling to faithfully convert between WGSL
+and “Shader” SPIR-V.
+By “faithful”, we mean that all salient properties of a shader should
+be preserved by a round trip conversion between the two ways of writing
+the shader: e.g. it should retain its semantics and its performance
+characteristics.
+Furthermore, those WebGPU Community Group members plan to add an option
```

Comment:
Again, not in the purview of the standardization group. If one member company wants to do something outside the bounds of the standardization group, they can go right ahead. But it isn't relevant to this repo.

Reply:
As above, the question comes up, so it is relevant to the FAQ, and what given members are thinking is useful information.

Reply:
> But it isn't relevant to this repo.

True, but in the end we'll just end up with every big participant to engage in an info war by releasing blog posts, where we'd have even less collaboration and mutual agreement on what's being announced...

@litherum do you disagree with the intention here? You don't want to have the existing compilers target WGSL?
If not, we could rephrase this as something like "WebGPU CG members anticipate ..." or "expect that ... will emerge" without mentioning who exactly is going to work on that.

Reply:
Having compilers target this language would be useful and valuable, regardless of the source language. Specifically mentioning a project by one member company which uses this language as a destination language, is irrelevant, and potentially even misleading.

Reply:
I didn't mention any specific projects, or particular member companies. I believe I already provided caveats and qualified language as @kvark suggested.
I mentioned an ecosystem that is highly relevant going forward, and was the proposed technology prior to the decision (which makes the question highly relevant).

Hey @litherum, would you prefer if I prefaced this section with a prominent notice along the lines of: "This section is outside the purview of the community group" ?